src/main/java/pl/med/clinic/service/VisitServiceImpl.java

```diff
@@ -15,11 +18,38 @@
 public class VisitServiceImpl implements VisitService {
 
     private final VisitRepository visitRepository;
+    private final PatientRepository patientRepository;
 
     @Override
-    public VisitsDtoResponse get(Long patientId) {
+    public VisitsDtoResponse getPatientsVisits(Long patientId) {
         List<VisitEntity> visits = visitRepository.getVisitEntitiesByPatientId(patientId);
         return new VisitsDtoResponse(visits);
     }
 
+    @Override
+    public VisitDto getVisit(Long visitId) {
+        return new VisitDto(visitRepository.getOrThrow(visitId));
+    }
+
+    @Override
+    public VisitDto createVisit(VisitDtoRequest newVisit) {
+        VisitEntity visitEntity = mapToEntity(newVisit);
+        VisitEntity save = visitRepository.save(visitEntity);
+        return new VisitDto(save);
+    }
+
+    @Override
+    public void cancelVisit(Long visitId) {
+        visitRepository.deleteById(visitId);
```

Comment:
usuwasz visit a czy sprawdzałaś czy zniknie ona z relacji pacjenta ??

Reply:
chodzi o to, że w przypadku kiedy nie ma żadnej wizyty pacjent też ma być skasowany? 
nie rozumiem, znikania z relacji pacjenta

Reply:
dodane - czy prawidłow, czy powinnam do tego napisać jakieś testy, jak to przetestować, rzecz poza tabelą - do przegadania

Reply:
do przegadania

Reply:
poprawione wspólnie